test/e2e/common/util_mayastor_crds.go

```diff
@@ -318,3 +330,73 @@ func CheckForMSVs() (bool, error) {
 	}
 	return foundResources, err
 }
+
+func CheckAllMsvsAreHealthy() error {
+	msvGVR := schema.GroupVersionResource{
+		Group:    "openebs.io",
+		Version:  "v1alpha1",
+		Resource: "mayastorvolumes",
+	}
+
+	allHealthy := true
+	retrieveErrors := false
+	msvs, err := gTestEnv.DynamicClient.Resource(msvGVR).Namespace(NSMayastor).List(context.TODO(), metav1.ListOptions{})
+	if err == nil && msvs != nil && len(msvs.Items) != 0 {
+		for _, msv := range msvs.Items {
+			msvName, _ := retrieveFieldStringValue(&msv, "metadata", "name")
+			state, err := retrieveFieldStringValue(&msv, "status", "state")
+			if err == nil {
+				if state != "healthy" {
+					logf.Log.Info("CheckAllMsvsAreHealthy", "msvName", msvName, "state", state)
+					allHealthy = false
```

Comment:
Couldn't we just return here? Given we are checking all are healthy, can't we return early as soon as we find the first failure?

Reply:
We could, but I wanted the test logs to show the complete set of unhealthy mayastor volumes. That may be useful when diagnosing issues, for example all msvs on a particular node may be unhealthy.

Reply:
Fair enough.